sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerContext.java

```diff
@@ -508,6 +509,9 @@ public String getPlanningError()
    */
   public void setPlanningError(String formatText, Object... arguments)
   {
+    if (queryContext().isDecoupledMode()) {
+      throw InvalidSqlInput.exception(formatText, arguments);
+    }
```

Comment:
Can we not have a "DecoupledPlannerContext" or something that just overrides this method and throws the exception instead of checks the context?  I don't actually know where the PlannerContext is created, so no clue how hard/easy it is to do it, but this check feels ugly.

Reply:
I feel like this whole class should be removed

Reply:
`queryContext` is passed as a `Map` instead of `QueryContext` 
I'll do this as a separate change